src/common/src/types/interval.rs

```diff
@@ -65,22 +66,160 @@ impl Interval {
+    /// ```
+    pub fn usecs_of_day(&self) -> u64 {
         self.usecs.rem_euclid(USECS_PER_DAY) as u64
     }
 
+    /// Returns the years field. range: unlimited
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "2332 yrs 12 months".parse().unwrap();
+    /// assert_eq!(interval.years_field(), 2333);
+    /// ```
+    pub fn years_field(&self) -> i32 {
+        self.months / 12
+    }
+
+    /// Returns the months field. range: `-11..=11`
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "15 months".parse().unwrap();
+    /// assert_eq!(interval.months_field(), 3);
+    ///
+    /// let interval: Interval = "-15 months".parse().unwrap();
+    /// assert_eq!(interval.months_field(), -3);
+    /// ```
+    pub fn months_field(&self) -> i32 {
+        self.months % 12
+    }
+
+    /// Returns the days field. range: unlimited
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "1 months 100 days 25:00:00".parse().unwrap();
+    /// assert_eq!(interval.days_field(), 100);
+    /// ```
+    pub fn days_field(&self) -> i32 {
+        self.days
+    }
+
+    /// Returns the hours field. range: unlimited
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "25:00:00".parse().unwrap();
+    /// assert_eq!(interval.hours_field(), 25);
+    ///
+    /// let interval: Interval = "-25:00:00".parse().unwrap();
+    /// assert_eq!(interval.hours_field(), -25);
+    /// ```
+    pub fn hours_field(&self) -> i32 {
+        (self.usecs / USECS_PER_SEC / 3600) as i32
+    }
+
+    /// Returns the minutes field. range: `-59..=-59`
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "00:20:00".parse().unwrap();
+    /// assert_eq!(interval.minutes_field(), 20);
+    ///
+    /// let interval: Interval = "-00:20:00".parse().unwrap();
+    /// assert_eq!(interval.minutes_field(), -20);
+    /// ```
+    pub fn minutes_field(&self) -> i32 {
+        (self.usecs / USECS_PER_SEC / 60 % 60) as i32
+    }
+
+    /// Returns the seconds field, including fractional parts, in microseconds.
+    /// range: `-59_999_999..=59_999_999`
+    ///
+    /// # Example
+    /// ```
+    /// # use risingwave_common::types::Interval;
+    /// let interval: Interval = "01:02:03.45678".parse().unwrap();
+    /// assert_eq!(interval.seconds_in_micros(), 3_456_780);
+    ///
+    /// let interval: Interval = "-01:02:03.45678".parse().unwrap();
+    /// assert_eq!(interval.seconds_in_micros(), -3_456_780);
+    /// ```
+    pub fn seconds_in_micros(&self) -> i32 {
+        (self.usecs % (USECS_PER_SEC * 60)) as i32
+    }
+
+    /// Returns the microseconds since 1970-01-01 00:00:00+00.
```

Comment:
The total number of microseconds, as defined by PostgreSQL `extract`. Also note this value is not used by interval ordering (IntervalCmpValue) and is not consistent with it. (the "wired behavior" comment below)